doc/assy.rst

```diff
@@ -0,0 +1,359 @@
+
+.. code-block:: python
+   :linenos:
+
+    def make_vslot(l):
+    
+        return PROFILE.toPending().extrude(l)
+    
+    
+    def make_connector():
+    
+        rv = (
+            cq.Workplane()
+            .box(20, 20, 20)
+            .faces("<X")
+            .workplane()
+            .cboreHole(6, 15, 18)
+            .faces("<Z")
+            .workplane()
+            .cboreHole(6, 15, 18)
+        )
+    
+        # tag mating faces
+        rv.faces(">X").tag("X").end()
+        rv.faces(">Z").tag("Z").end()
+    
+        return rv
+    
+    
+    def make_panel(w, h, t, cutout):
+    
+        rv = (
+            cq.Workplane("XZ")
+            .rect(w, h)
+            .extrude(t)
+            .faces(">Y")
+            .vertices()
+            .rect(2*cutout,2*cutout)
+            .cutThruAll()
+            .faces("<Y")
+            .workplane()
+            .pushPoints([(-w / 3, HANDLE_L / 2), (-w / 3, -HANDLE_L / 2)])
+            .hole(3)
+        )
+    
+        # tag mating edges
+        rv.faces(">Y").edges("%CIRCLE").edges(">Z").tag("hole1")
+        rv.faces(">Y").edges("%CIRCLE").edges("<Z").tag("hole2")
+    
+        return rv
+    
+    
+    def make_handle(w, h, r):
+    
+        pts = ((0, 0), (w, 0), (w, h), (0, h))
+    
+        path = cq.Workplane().polyline(pts)
+    
+        rv = (
+            cq.Workplane("YZ")
+            .rect(r, r)
+            .sweep(path, transition="round")
+            .tag("solid")
+            .faces("<X")
+            .workplane()
+            .faces("<X", tag="solid")
+            .hole(r / 1.5)
+        )
+        
+        # tag mating faces
+        rv.faces("<X").faces(">Y").tag("mate1")
+        rv.faces("<X").faces("<Y").tag("mate2")
+    
+        return rv
+        
+Initial assembly
+================
+
+Next we want to instantiate all the components and add them to the assembly.
+
+.. code-block:: python
+  :linenos:
+   
+    # define the elements
+    door = (
+        cq.Assembly()
+        .add(make_vslot(H), name="left")
+        .add(make_vslot(H), name="right")
+        .add(make_vslot(W), name="top")
+        .add(make_vslot(W), name="bottom")
+        .add(make_connector(), name="con_tl", color=cq.Color("black"))
+        .add(make_connector(), name="con_tr", color=cq.Color("black"))
+        .add(make_connector(), name="con_bl", color=cq.Color("black"))
+        .add(make_connector(), name="con_br", color=cq.Color("black"))
+        .add(
+            make_panel(W + SLOT_D, H + SLOT_D, PANEL_T),
+            name="panel",
+            color=cq.Color(0, 0, 1, 0.2),
+        )
+        .add(
+            make_handle(HANDLE_D, HANDLE_L, HANDLE_W),
+            name="handle",
+            color=cq.Color("yellow"),
+        )
+    )
+    
+Constraints definition
+======================
+
+Then we want to define all the constraints
+
+.. code-block:: python
+  :linenos:
+   
+    # define the elements
+    door = (
```

Comment:
I think you've cut and pasted the wrong block, this is the same as the previous section.

Reply:
Good catch, solved